components/match2/commons/starcraft_starcraft2/match_group_input_starcraft.lua

```diff
@@ -374,73 +373,50 @@ end
 
-	---@type {input: string, faction: string?, link: string?}[]
-	local participantsList = Array.mapIndexes(function(playerIndex)
-		local prefix = 't' .. opponentIndex .. 'p' .. playerIndex
-
-		if Logic.isEmpty(mapInput[prefix]) then return end
-
-		return {
-			input = mapInput[prefix],
-			link = Logic.nilIfEmpty(mapInput[prefix .. 'link']),
-			faction = isArchon and archonFaction or Faction.read(mapInput[prefix .. 'race']),
-		}
+	local players = Array.mapIndexes(function(playerIndex)
+		return Logic.nilIfEmpty(mapInput['t' .. opponentIndex .. 'p' .. playerIndex])
 	end)
 
-	local participants = {}
-
-	Array.forEach(participantsList, function(participantInput, position)
-		local nameInput = participantInput.input
-
-		local isTBD = nameInput:upper() == TBD or nameInput:upper() == TBA
-
-		local link = participantInput.link or Variables.varDefault(nameInput .. '_page') or nameInput
-		link = Page.pageifyLink(link) --[[@as string -- can't be nil as input isn't nil]]
-
-		local playerIndex = MapFunctions.getPlayerIndex(players, link, nameInput)
-
-		-- in case we have a TBD or a player not known in match2players inster a new player in match2players
-		if isTBD or playerIndex == 0 then
-			table.insert(players, {
-				name = isTBD and TBD or link,
-				displayname = isTBD and TBD or nameInput,
-				extradata = {faction = participantInput.faction or Faction.defaultFaction},
-			})
-			playerIndex = #players
+	local participants, unattachedParticipants = MatchGroupInputUtil.parseParticipants(
+		opponent.match2players,
+		players,
+		function(playerIndex)
+			local prefix = 't' .. opponentIndex .. 'p' .. playerIndex
+			return {
+				name = mapInput[prefix],
+				link = Logic.nilIfEmpty(mapInput[prefix .. 'link']) or Variables.varDefault(mapInput[prefix] .. '_page'),
+				faction = isArchon and archonFaction or Faction.read(mapInput[prefix .. 'race']),
+			}
+		end,
+		function(playerIndex, playerIdData, playerInputData)
+			return {
+				faction = playerInputData.faction or (playerIdData.extradata or {}).faction or Faction.defaultFaction,
+				player = playerIdData.name or playerInputData.link,
+				flag = Flags.CountryName(playerIdData.flag),
+				position = playerIndex,
+			}
 		end
+	)
 
-		local player = players[playerIndex]
-
-		participants[opponentIndex .. '_' .. playerIndex] = {
-			faction = participantInput.faction or player.extradata.faction,
-			player = link,
-			position = position,
-			flag = Flags.CountryName(player.flag),
-		}
+	Array.forEach(unattachedParticipants, function(participant)
+		local nameUpper = participant.name:upper()
+		local isTBD = nameUpper == TBD or nameUpper == TBA
+
+		table.insert(opponent.match2players, {
+			name = isTBD and TBD or participant.link,
+			displayname = isTBD and TBD or nameUpper,
+			flag = participant.flag,
+			extradata = {faction = participant.faction},
+		})
+		participants[#opponent.match2players] = participant
```

Comment:
does not work

would need both `toStoreData` and `playerInputData` accessible
the first has the data needed for the participant entry, the laster the one for setting up the match2player